drivers/source/UARTSerial.cpp

```diff
@@ -377,6 +377,7 @@ int UARTSerial::enable_input(bool enabled)
 {
     api_lock();
     SerialBase::enable_input(enabled);
+    enable_rx_irq(); // Enable interrupt to handle incoming data
```

Comment:
Yeah, covers a hole. I think when I originally did this I was viewing your test case -  trying to write to the device while it's disabled - as undefined behaviour. It was only intended as a power-saving optimisation for when you're not needing data, not a robust "mute".

But people are clearly finding use beyond my original plan, so may as well make it solid. And there's nothing in the docs saying you shouldn't do what your test is doing.

Note that this doesn't *disable* the IRQ when doing `enable_input(false)`. Maybe it should? Should your `enable_rx_irq()` routine actually be a `enable_or_disable_irq()`?

I can imagine there are other users who think that `enable_input(false)` should reliably stop rx and tx interrupts from being generated.

Maybe not within the scope of this PR though.

Reply:
I tried to keep the changes least intrusive, while fixing the described issue. I think each change will be detected by users especially if `UARTSerial::read` or `UARTSerial::write` or the interrupt routines are changed in any way. This is why I would prefer to avoid changing them and using some `enable_or_disable_irq` there instead.

Eventually my renaming proposal in the previous comment is sufficient?

Reply:
True, making it mute more effectively might break someone's use case. No matter what they might be expecting, they clearly they can't be relying on what it isn't doing.